api/resources_portal/views/material_request.py

```diff
@@ -488,17 +488,53 @@ def update(self, request, *args, **kwargs):
         material_request.refresh_from_db()
 
         if request.user == material_request.requester:
-            if field_changed("irb_attachment", original_material_request, material_request):
+            irb_changed = field_changed(
+                "irb_attachment", original_material_request, material_request
+            )
+            mta_changed = field_changed(
+                "requester_signed_mta_attachment", original_material_request, material_request
+            )
+            payment_changed = field_changed(
+                "payment_method", original_material_request, material_request
+            )
+            payment_notes_changed = field_changed(
+                "payment_notes_method", original_material_request, material_request
+            )
+
+            # Don't allow piecemeal uploads of required documents
+            # because they make determining the state of the request
+            # difficult. If someone is uploading docs, they need to
+            # upload everything at once.
+            if irb_changed or mta_changed or payment_changed or payment_notes_changed:
+                missing_document = False
+                if original_material_request.material.needs_irb and not irb_changed:
+                    missing_document = True
+                if original_material_request.material.needs_mta and not mta_changed:
+                    missing_document = True
+                if (
+                    original_material_request.material.shipping_requirement
+                    and original_material_request.material.shipping_requirement.needs_payment
+                ):
+                    if not (payment_changed or original_material_request.payment_method) or (
+                        payment_notes_changed or original_material_request.payment_notes_method
+                    ):
```

Comment:
I think that the original_material_request should always have these be empty at least it should be if we are enforcing these changes.

Reply:
I agree, but we haven't until now and this just seems overall safer than assuming they are.